[18.0][ADD] rma_sale_stock_restocking_fee_invoicing

sbejaoui opened the pull request:
This module extends the standard RMA flow and the behavior of
`sale_stock_restocking_fee_invoicing` by allowing:

- Fixed or percentage-based restocking fees.
- Automatic fee application during RMA receipt.
- Integration with different refund strategies:
  - Update sale order quantity.
  - Manual refund after receipt.

needs: https://github.com/OCA/stock-logistics-workflow/pull/2150

sbejaoui committed:
[ADD] rma_sale_stock_restocking_fee_invoicing

This module extends the standard RMA flow and the behavior of
`sale_stock_restocking_fee_invoicing` by allowing:

- Fixed or percentage-based restocking fees.
- Automatic fee application during RMA receipt.
- Integration with different refund strategies:
  - Update sale order quantity.
  - Manual refund after receipt.

sbejaoui committed:
[DON'T MERGE] add test-requirements.txt

marielejeune commented on rma_sale_stock_restocking_fee_invoicing/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+
+{
+    "name": "Rma Sale Stock Restocking Fee Invoicing",
+    "summary": """c""",
```
Please include a summary

marielejeune commented on rma_sale_stock_restocking_fee_invoicing/models/rma_operation.py:
```diff
@@ -0,0 +1,44 @@
+        ],
+        default="no_fee",
+        help="Define whether the restocking fee is a fixed amount or a percentage.",
+    )
```
I would add a `required=True` attribute, since in the methods you check that `restocking_fee_type != 'no_fee'`, you want to avoid false values.

sbejaoui replied:
I removed the 'no_fee' option

marielejeune reviewed (changes requested):
LGTM, 2 minor comments

sbejaoui committed:
[IMP] rma_sale_stock_restocking_fee_invoicing: allow to input restocking fee on rma

sbejaoui committed:
[FIX] rma_sale_stock_restocking_fee_invoicing: price is based on sale line

sbejaoui committed:
[FIX] rma_sale_stock_restocking_fee_invoicing: create invoice at reception

sbejaoui committed:
[IMP] rma_sale_stock_restocking_fee_invoicing: manage different UoM & use sale line price_unit

marielejeune committed:
[DONT MERGE] test-requirements

marielejeune committed:
[IMP] rma_sale_stock_restocking_fee_invoicing: multi-step return process
Charge fees at the last step of the return process in case of
a multi-step return route.

sbejaoui committed:
Merge pull request #4 from acsone/18.0-rma_sale_stock_restocking_fee_invoicing-imp-mle

[IMP] rma_sale_stock_restocking_fee_invoicing: multi-step return route - Charge at last step
